sled-agent/config-reconciler/src/dataset_serialization_task.rs

```diff
@@ -1849,3 +1890,655 @@ mod tests {
         logctx.cleanup_successful();
     }
 }
+
+// On illumos, run a set of tests that use the `RealZfs` implementation on
+// zpools backed by vdevs.
+#[cfg(all(test, target_os = "illumos"))]
+mod illumos_tests {
```

Comment:
Thanks for doing this -- hopefully it's kept pretty "local to this task" so the cost of keeping this test harness around isn't painful, but it's reassuring to have this multi-layered testing approach.